packages/contracts/src/dollar/SushiSwapPool.sol

```diff
@@ -1,25 +1,25 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.3;
+pragma solidity 0.8.16;
 
 import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
 import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol";
-import "./UbiquityAlgorithmicDollarManager.sol";
+import "./core/UbiquityDollarManager.sol";
 
 contract SushiSwapPool {
-    IUniswapV2Factory public factory =
+    IUniswapV2Factory constant factory =
         IUniswapV2Factory(0xC0AEe478e3658e2610c5F7A4A2E1777cE9e4f2Ac);
```

Comment:
I am just wondering if we need this contract. If there's no dependency for SushiSwapPool.sol, lets remove it! If we go to optimism, it will definitely not work bc of hardcoded factory address at least.

Reply:
It's used for launching the pool. We'd have to write a different version for Optimism which is standard

Reply:
And yes, it'll break the UI.

Reply:
ok I changed base branch from development from contract-development.